libpromises/loading.c

```diff
@@ -481,22 +493,32 @@ static JsonElement *ReadReleaseIdFileFromInputs()
 
 Policy *LoadPolicy(EvalContext *ctx, GenericAgentConfig *config)
 {
-    StringSet *parsed_files_and_checksums = StringSetNew();
+    StringMap *policy_files_hashes = StringMapNew();
+    StringSet *parsed_files_checksums = StringSetNew();
     StringSet *failed_files = StringSetNew();
 
     Banner("Loading policy");
 
     Policy *policy = LoadPolicyFile(ctx, config, config->input_file,
-                                    parsed_files_and_checksums, failed_files);
+                                    policy_files_hashes, parsed_files_checksums,
+                                    failed_files);
 
     if (StringSetSize(failed_files) > 0)
     {
         Log(LOG_LEVEL_ERR, "There are syntax errors in policy files");
         DoCleanupAndExit(EXIT_FAILURE);
     }
 
-    StringSetDestroy(parsed_files_and_checksums);
+    StringSetDestroy(parsed_files_checksums);
     StringSetDestroy(failed_files);
+    if (policy != NULL)
+    {
+        policy->policy_files_hashes = policy_files_hashes;
```

Comment:
Because I don't want all the `Policy` objects/structs created in the process of loading policy to have a reference to this map. Only the final policy which is the result of the whole `LoadPolicy()` run should have it. `LoadPolicyFile()` is called recursively in the process of loading policy and individual "sub-policies" are merged into the final "big policy" which gets the reference to the map in the end.